server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -145,8 +145,10 @@ public class InternalEngine extends Engine {
     private final LiveVersionMapArchive liveVersionMapArchive;
     // Records the last known generation during which LiveVersionMap was in unsafe mode. This indicates that only after this
     // generation it is safe to rely on the LiveVersionMap for a real-time get.
-    // TODO: can we move this entirely into the stateless plugin?
+    // TODO: move the following two to the stateless plugin
```

Comment:
I think, extending the `LiveVersionMapArchive` interface to have a `preCommit` and `switchToSafeMap` hook, is probably one way of moving these out of here.

Reply:
That makes sense to me.